src/ExperimentalInternetComputer.mo

```diff
@@ -55,6 +55,26 @@ module {
     // performance_counter costs around 200 extra instructions, we perform an empty measurement to decide the overhead
     let overhead = pre - init;
     post - pre - overhead
-  }
+  };
+
+  /// The same as `countInstructions`, but accepts async* functions
+  ///
+  /// Example:
+  /// ```motoko no-repl
+  /// import IC "mo:base/ExperimentalInternetComputer";
+  ///
+  /// let count = await* IC.countInstructionsAsync(func(): async* () {
+  ///   // ...
+  /// });
+  /// ```
+  public func countInstructionsAsync(comp : () -> async* ()) : async* Nat64 {
+    let init = Prim.performanceCounter(0);
+    let pre = Prim.performanceCounter(0);
+    await* comp();
```

Comment:
The problem with this approach is that when the `comp` callback decides to do an `async` call the `performanceCounter(0)` will be reset to 0 🫤. There is an upcoming feature proposal that will preserve another `performanceCounter` across those call contexts and may do what you want...

Reply:
This is probably available now as performanceCounter(1), right @ggreif ?